submodules-tasks.yml

```diff
@@ -0,0 +1,364 @@
+version: "3"
+vars:
+    G_DEP_DOWNLOAD_SCRIPT: "{{.ROOT_DIR}}/tools/scripts/deps-download/download-dep.py"
+
+tasks:
+  default:
+    cmds:
+      - task: "core"
+      - task: "log-viewer"
+
+  log-viewer:
+    cmds:
+      - task: "yscope-log-viewer"
+
+  core:
+    cmds:
+      - task: "abseil-cpp"
+      - task: "antlr4"
+      - task: "Catch2"
+      - task: "date"
+      - task: "json"
+      - task: "log-surgeon"
+      - task: "outcome"
+      - task: "simdjson"
+      - task: "sqlite3"
+      - task: "yaml-cpp"
+
+  abseil-cpp:
+    internal: true
+    vars:
+      CHECKSUM_FILE: "{{.G_BUILD_DIR}}/{{.TASK | replace \":\" \"#\"}}.md5"
+      DESTINATION: "{{.G_CORE_SUBMODULE_DIR}}/abseil-cpp"
+    sources:
+      - "{{.G_DEP_DOWNLOAD_SCRIPT}}"
+      - "{{.TASKFILE}}"
+    generates: [ "{{.CHECKSUM_FILE}}" ]
+    deps:
+      - ":init"
+      - task: ":utils:validate-checksum"
+        vars:
+          CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
+          DATA_DIR: "{{.DESTINATION}}"
+    cmds:
+      - task: "download-dependency"
+        vars:
+          CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
```

Comment:
* `CHECKSUM_FILE` is no longer an argument.
* Let's alphabetize the parameters.